fotoapparat/src/main/java/io/fotoapparat/routine/focus/ManualFocusRoutine.java

```diff
@@ -0,0 +1,45 @@
+package io.fotoapparat.routine.focus;
+
+import android.graphics.Rect;
+
+import java.util.concurrent.Executor;
+
+import io.fotoapparat.hardware.CameraDevice;
+import io.fotoapparat.result.FocusResult;
+import io.fotoapparat.result.PendingResult;
+import io.fotoapparat.view.TapToFocusSupporter;
+
+public class ManualFocusRoutine implements TapToFocusSupporter.FocusCallback {
+
+    private final CameraDevice cameraDevice;
+    private final Executor cameraExecutor;
+
+    private PendingResult<FocusResult> manualFocusResult;
+
+    public ManualFocusRoutine(CameraDevice cameraDevice,
+                              Executor cameraExecutor) {
+        this.cameraDevice = cameraDevice;
+        this.cameraExecutor = cameraExecutor;
+    }
+
+    public PendingResult<FocusResult> manualFocus(Rect cameraViewRect, float focusX, float focusY) {
+        ManualFocusTask manualFocusTask = new ManualFocusTask(cameraDevice, cameraViewRect, focusX, focusY);
+        cameraExecutor.execute(manualFocusTask);
+
+        return PendingResult.fromFuture(manualFocusTask);
+    }
+
+    @Override
+    public void onManualFocus(final Rect cameraViewRect, float focusX, float focusY) {
```

Comment:
This callback should not be here. Routine is only responsible for performing the "business logic" on camera (exactly what your `manualFocus` method does). Reacting on taps is a responsibility of the view.